Get rid of useless anonymous classes

hasys opened the pull request:
Hi @wmedvede, @evacchi, @LuboTerifaj,

I move all changes (add added some more) related to Maps, Lists and Sets builders to this PR. When this PR will be merged inner classes amount will be reduced ~35.

wmedvede commented on kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/marshall/json/oryx/BaseOryxIdMappings.java:
```diff
@@ -152,110 +153,62 @@ public void addOryxDefinitionId(final Class<?> type,
-
-        return globalMappings;
+
+        return new Maps.Builder<Class<?>, String>()
```
would be nice to keep a instance field instead of returning a fresh new calculated on each envocation

wmedvede commented on kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/marshall/json/oryx/BaseOryxIdMappings.java:
```diff
@@ -266,215 +219,122 @@ public void addOryxDefinitionId(final Class<?> type,
-        }};
-
-        return skippedProperties;
+        return new Maps.Builder<Class<?>, Set<String>>()
```
same here

wmedvede commented on kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/marshall/json/oryx/BaseOryxIdMappings.java:
```diff
@@ -576,16 +434,6 @@ public String getDefinitionId(final Class<?> clazz) {
         return BindableAdapterUtils.getDefinitionId(clazz);
     }
 
-    private Class<?> get(final String oryxId,
```
Nice removal!

wmedvede commented on kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/workitem/WorkItemDefinitionParser.java:
```diff
@@ -280,9 +280,8 @@ private static void set(final Map<String, Object> map,
                             final String defaultValue,
                             final Consumer<String> consumer) {
         final String value = (String) map.get(key);
-        if (!isEmpty(value)) {
```
isEmpty is probably more consistent with other equivalent methods like the ones in Collections.isEmpty(), etc.
Just a personal preference

hasys replied:
@wmedvede yes, but this method is private and all invocations of this method are reversed. So, to me it looks like the goal of this method is check that value is **non** empty. That's why I changed it. 

But yes, both approaches has it's points. Not sure which one is better in this situation.

triceo replied:
Generally, I'd prefer `isEmpty()`. Two reasons:
- `isNotEmpty()` is easily doable as `!isEmpty()`, but 
-  in the other case `isEmpty()` would have to be represented as `!isNotEmpty()` and that doesn't read well.

hasys replied:
@triceo, make sense. I checked how similar situations are done in the Java standard library. And found good example: there are two methods in `Objects` class: `Objects.isNull` and `Objects.nonNull`. 

Since in our class we private method and both usages are `nonEmpty` I renamed it from `isNotEmpty` to `nonEmpty` I guess it makes sense in this situation and it is close to Java language approach.

jomarko commented:
@hasys do you have some task where you track this anonymous class replacement please?

hasys commented:
Jenkins execute full downstream build

hasys commented:
Great catch, @wmedvede,

I added `<inherits name="org.kie.soup.commons.KIESoupCommons"/>` to all modified client and API modules.

jomarko commented:
@hasys understood, asking just if your intention is to go through whole kie-wb-common codebase or should I help you with this.

hasys commented:
@jomarko "short" term goal for me - to fix everything in Stunner/Process Designer, probably CM Designer for now. But if I will found something really blocker/critical I will send PR with fixes to any project for sure. But total clean up for entire kie-wb-common is not the case for now :(

romartin commented:
missing @triceo 's approval for merging

romartin commented:
oh @hasys there are now conflicts, please lemme know once ready

hasys commented:
Hi @romartin,

rebased and pushed, thank you!

@manstis I added new method to your StringUtils: nonEmpty to have it in same way as for `Objects.isNull`/`Objects.nonNull` from standard Java library. In my opinion `nonEmpty(string)` is more readable than `!StringUtils.isEmpty(string)`.

hasys commented:
Jenkins execute full downstream build

hasys commented:
Hi @jhrcek,

not sure what happened here :(

<details>
  <summary>Regular Jenkins build</summary>
  <pre>13:55:30 [INFO] ------------------------------------------------------------------------
13:55:30 [INFO] BUILD FAILURE
13:55:30 [INFO] ------------------------------------------------------------------------
13:55:30 [INFO] Total time: 01:33 h
13:55:30 [INFO] Finished at: 2018-10-05T09:55:30-04:00
13:55:34 [INFO] Final Memory: 1111M/2878M
13:55:34 [INFO] ------------------------------------------------------------------------
13:55:34 [WARNING] The requested profile "wildfly11" could not be activated because it does not exist.
13:55:34 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.18.1:test (default-test) on project kie-wb-common-ala-docker-provider: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.18.1:test failed: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
13:55:34 [ERROR] Command was /bin/sh -c cd /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider && /opt/tools/jdk1.8.0_152/jre/bin/java -Xmx1024m -Dfile.encoding=UTF-8 -jar /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider/target/surefire/surefirebooter328349617163071388.jar /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider/target/surefire/surefire457629716105897865tmp /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider/target/surefire/surefire_1356356627607307371257tmp
13:55:34 [ERROR] -> [Help 1]
13:55:34 org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.18.1:test (default-test) on project kie-wb-common-ala-docker-provider: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.18.1:test failed: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
13:55:34 Command was /bin/sh -c cd /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider && /opt/tools/jdk1.8.0_152/jre/bin/java -Xmx1024m -Dfile.encoding=UTF-8 -jar /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider/target/surefire/surefirebooter328349617163071388.jar /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider/target/surefire/surefire457629716105897865tmp /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider/target/surefire/surefire_1356356627607307371257tmp
13:55:34     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
13:55:34     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
13:55:34     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
13:55:34     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
13:55:34     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
13:55:34     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
13:55:34     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
13:55:34     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
13:55:34     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
13:55:34     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
13:55:34     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
13:55:34     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
13:55:34     at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
13:55:34     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
13:55:34     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
13:55:34     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
13:55:34     at java.lang.reflect.Method.invoke (Method.java:498)
13:55:34     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
13:55:34     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
13:55:34     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
13:55:34     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
13:55:34 Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.18.1:test failed: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
13:55:34 Command was /bin/sh -c cd /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider && /opt/tools/jdk1.8.0_152/jre/bin/java -Xmx1024m -Dfile.encoding=UTF-8 -jar /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider/target/surefire/surefirebooter328349617163071388.jar /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider/target/surefire/surefire457629716105897865tmp /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider/target/surefire/surefire_1356356627607307371257tmp
13:55:34     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:145)
13:55:34     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
13:55:34     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
13:55:34     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
13:55:34     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
13:55:34     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
13:55:34     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
13:55:34     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
13:55:34     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
13:55:34     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
13:55:34     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
13:55:34     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
13:55:34     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
13:55:34     at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
13:55:34     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
13:55:34     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
13:55:34     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
13:55:34     at java.lang.reflect.Method.invoke (Method.java:498)
13:55:34     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
13:55:34     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
13:55:34     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
13:55:34     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
13:55:34 Caused by: java.lang.RuntimeException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
13:55:34 Command was /bin/sh -c cd /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider && /opt/tools/jdk1.8.0_152/jre/bin/java -Xmx1024m -Dfile.encoding=UTF-8 -jar /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider/target/surefire/surefirebooter328349617163071388.jar /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider/target/surefire/surefire457629716105897865tmp /home/jenkins/workspace/KIE/master/pullrequest/kie-wb-common-pullrequests/kie-wb-common-ala/kie-wb-common-ala-docker-provider/target/surefire/surefire_1356356627607307371257tmp
13:55:34     at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork (ForkStarter.java:515)
13:55:34     at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork (ForkStarter.java:380)
13:55:34     at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run (ForkStarter.java:167)
13:55:34     at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider (AbstractSurefireMojo.java:990)
13:55:34     at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked (AbstractSurefireMojo.java:824)
13:55:34     at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute (AbstractSurefireMojo.java:722)
13:55:34     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
13:55:34     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
13:55:34     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
13:55:34     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
13:55:34     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
13:55:34     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
13:55:34     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
13:55:34     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
13:55:34     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
13:55:34     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
13:55:34     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
13:55:34     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
13:55:34     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
13:55:34     at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
13:55:34     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
13:55:34     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
13:55:34     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
13:55:34     at java.lang.reflect.Method.invoke (Method.java:498)
13:55:34     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
13:55:34     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
13:55:34     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
13:55:34     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
13:55:34 [ERROR] 
13:55:34 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
13:55:34 [ERROR] 
13:55:34 [ERROR] For more information about the errors and possible solutions, please read the following articles:
13:55:34 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
13:55:34 [ERROR] 
13:55:34 [ERROR] After correcting the problems, you can resume the build with the command
13:55:34 [ERROR]   mvn <goals> -rf :kie-wb-common-ala-docker-provider
13:55:34 Build step 'Invoke top-level Maven targets' marked build as failure
13:55:36 Recording test results
13:56:04 Skipping publisher since build result is FAILURE
13:56:04 Skipping publisher since build result is FAILURE
13:56:04 Archiving artifacts
13:56:07 Adding one-line test results to commit status...
13:56:07 Setting status of f25aa8e28c1b0b7f0ac0d0bb734c13036839c5bc to FAILURE with url https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/kie-wb-common-pullrequests/536/ and message: 'Build finished. 10086 tests run, 69 skipped, 0 failed.'
13:56:07 Using context: Linux
13:56:07 [WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
13:56:07 Finished: FAILURE</pre>
</details>

<details>
  <summary>Full Downstream build</summary>
  <pre>14:46:09 [INFO] ------------------------------------------------------------------------
14:46:09 [INFO] Reactor Summary:
14:46:09 [INFO] 
14:46:09 [INFO] KIE Workbench - Distributions ...................... SUCCESS [  4.094 s]
14:46:09 [INFO] KIE Theme .......................................... SUCCESS [  0.131 s]
14:46:09 [INFO] KIE Theme - Community .............................. SUCCESS [  4.696 s]
14:46:09 [INFO] KIE Theme - Product ................................ SUCCESS [  1.201 s]
14:46:09 [INFO] KIE Workbench - Webapp Common ...................... SUCCESS [  0.603 s]
14:46:09 [INFO] KIE Workbench Deployment Validation ................ SUCCESS [  7.568 s]
14:46:09 [INFO] KIE Workbench Parent ............................... SUCCESS [  0.122 s]
14:46:09 [INFO] KIE Workbench - Home Page - Community .............. SUCCESS [ 30.585 s]
14:46:09 [INFO] KIE Workbench - Monitoring Webapp .................. SUCCESS [13:53 min]
14:46:09 [INFO] KIE Workbench - Webapp ............................. SUCCESS [19:58 min]
14:46:09 [INFO] KIE Workbench - Distribution Wars Parent ........... SUCCESS [  1.912 s]
14:46:09 [INFO] KIE Workbench - Monitoring Distribution Wars ....... SUCCESS [ 25.382 s]
14:46:09 [INFO] KIE Workbench - Distribution Wars .................. SUCCESS [03:15 min]
14:46:09 [INFO] KIE Drools Workbench Parent ........................ SUCCESS [  0.104 s]
14:46:09 [INFO] KIE Drools Workbench - Home Page - Community ....... SUCCESS [  8.896 s]
14:46:09 [INFO] KIE Drools Workbench - Webapp ...................... SUCCESS [20:23 min]
14:46:09 [INFO] KIE Drools Workbench - Distribution Wars ........... SUCCESS [ 44.077 s]
14:46:09 [INFO] KIE (Drools) Workbench Tests :: Parent ............. SUCCESS [  2.680 s]
14:46:09 [INFO] KIE (Drools) Workbench Tests :: REST API ........... FAILURE [03:18 min]
14:46:09 [INFO] KIE (Drools) Workbench Tests :: GUI ................ SKIPPED
14:46:09 [INFO] KIE Server Distributions ........................... SUCCESS [  0.758 s]
14:46:09 [INFO] jBPM server distribution ........................... SUCCESS [ 16.547 s]
14:46:09 [INFO] ------------------------------------------------------------------------
14:46:09 [INFO] BUILD FAILURE
14:46:09 [INFO] ------------------------------------------------------------------------
14:46:09 [INFO] Total time: 01:03 h
14:46:09 [INFO] Finished at: 2018-10-05T10:46:09-04:00
14:46:11 [INFO] [talledLocalContainer] 10:46:09,680 INFO  [org.jboss.errai.bus.server.cluster.noop.NoopClusteringProvider] (ServerService Thread Pool -- 79) clustering support not configured.
14:46:11 [INFO] Final Memory: 627M/2471M
14:46:11 [INFO] ------------------------------------------------------------------------
14:46:11 [INFO] [talledLocalContainer] 10:46:09,694 INFO  [org.jboss.errai.bus.server.service.bootstrap.OrderedBootstrap] (ServerService Thread Pool -- 79) starting errai bus ...
14:46:11 [INFO] [talledLocalContainer] 10:46:09,696 INFO  [org.jboss.errai.bus.server.service.bootstrap.DefaultComponents] (ServerService Thread Pool -- 79) using dispatcher implementation: org.jboss.errai.bus.server.SimpleDispatcher
14:46:11 [INFO] [talledLocalContainer] 10:46:09,699 INFO  [org.jboss.errai.bus.server.service.bootstrap.DefaultComponents] (ServerService Thread Pool -- 79) using session provider implementation: org.jboss.errai.bus.server.HttpSessionProvider
14:46:11 [WARNING] The requested profile "no-showcase" could not be activated because it does not exist.
14:46:11 [INFO] [talledLocalContainer] 10:46:09,727 INFO  [org.jboss.errai.bus.server.service.bootstrap.LoadExtensions] (ServerService Thread Pool -- 79) searching for errai extensions ...
14:46:11 [INFO] [talledLocalContainer] 10:46:09,779 INFO  [org.jboss.errai.bus.server.service.bootstrap.OrderedBootstrap] (ServerService Thread Pool -- 79) errai bus started.
14:46:11 [INFO] [talledLocalContainer] 10:46:09,785 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 79) WFLYUT0021: Registered web context: '/kie-wb' for server 'default-server'
14:46:11 [INFO] [talledLocalContainer] 10:46:09,808 INFO  [org.jboss.as.server] (ServerService Thread Pool -- 37) WFLYSRV0010: Deployed "kie-wb.war" (runtime-name : "kie-wb.war")
14:46:11 [INFO] [talledLocalContainer] 10:46:09,892 INFO  [org.jboss.as.server] (Controller Boot Thread) WFLYSRV0212: Resuming server
14:46:11 [INFO] [talledLocalContainer] 10:46:09,915 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0060: Http management interface listening on http://127.0.0.1:46810/management
14:46:11 [INFO] [talledLocalContainer] 10:46:09,915 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0051: Admin console listening on http://127.0.0.1:46810
14:46:11 [INFO] [talledLocalContainer] 10:46:09,915 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0025: WildFly Full 11.0.0.Final (WildFly Core 3.0.8.Final) started in 152121ms - Started 814 of 1040 services (353 services are lazy, passive or on-demand)
14:46:11 [ERROR] Failed to execute goal org.codehaus.cargo:cargo-maven2-plugin:1.6.9:start (start-container) on project kie-wb-tests-rest: Cannot start container [org.codehaus.cargo.container.wildfly.WildFly11xInstalledLocalContainer@4a8ab87d]: Monitor [org.codehaus.cargo.container.spi.startup.CombinedContainerMonitor] failed to detect running container within the timeout period [120000]. -> [Help 1]
14:46:11 org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.cargo:cargo-maven2-plugin:1.6.9:start (start-container) on project kie-wb-tests-rest: Cannot start container [org.codehaus.cargo.container.wildfly.WildFly11xInstalledLocalContainer@4a8ab87d]
14:46:11     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
14:46:11     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
14:46:11     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
14:46:11     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
14:46:11     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
14:46:11     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
14:46:11     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
14:46:11     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
14:46:11     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
14:46:11     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
14:46:11     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
14:46:11     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
14:46:11     at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
14:46:11     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
14:46:11     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
14:46:11     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
14:46:11     at java.lang.reflect.Method.invoke (Method.java:498)
14:46:11     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
14:46:11     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
14:46:11     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
14:46:11     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
14:46:11 Caused by: org.apache.maven.plugin.MojoExecutionException: Cannot start container [org.codehaus.cargo.container.wildfly.WildFly11xInstalledLocalContainer@4a8ab87d]
14:46:11     at org.codehaus.cargo.maven2.ContainerStartMojo.doExecute (ContainerStartMojo.java:74)
14:46:11     at org.codehaus.cargo.maven2.AbstractCargoMojo.execute (AbstractCargoMojo.java:471)
14:46:11     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
14:46:11     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
14:46:11     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
14:46:11     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
14:46:11     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
14:46:11     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
14:46:11     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
14:46:11     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
14:46:11     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
14:46:11     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
14:46:11     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
14:46:11     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
14:46:11     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
14:46:11     at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
14:46:11     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
14:46:11     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
14:46:11     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
14:46:11     at java.lang.reflect.Method.invoke (Method.java:498)
14:46:11     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
14:46:11     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
14:46:11     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
14:46:11     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
14:46:11 Caused by: org.codehaus.cargo.container.ContainerException: Monitor [org.codehaus.cargo.container.spi.startup.CombinedContainerMonitor] failed to detect running container within the timeout period [120000].
14:46:11     at org.codehaus.cargo.container.spi.AbstractLocalContainer.waitForStarting (AbstractLocalContainer.java:352)
14:46:11     at org.codehaus.cargo.container.wildfly.internal.AbstractWildFlyInstalledLocalContainer.waitForCompletion (AbstractWildFlyInstalledLocalContainer.java:383)
14:46:11     at org.codehaus.cargo.container.spi.AbstractLocalContainer.start (AbstractLocalContainer.java:232)
14:46:11     at org.codehaus.cargo.maven2.ContainerStartMojo.executeLocalContainerAction (ContainerStartMojo.java:84)
14:46:11     at org.codehaus.cargo.maven2.ContainerStartMojo.doExecute (ContainerStartMojo.java:58)
14:46:11     at org.codehaus.cargo.maven2.AbstractCargoMojo.execute (AbstractCargoMojo.java:471)
14:46:11     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
14:46:11     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
14:46:11     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
14:46:11     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
14:46:11     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
14:46:11     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
14:46:11     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
14:46:11     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
14:46:11     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
14:46:11     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
14:46:11     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
14:46:11     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
14:46:11     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
14:46:11     at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
14:46:11     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
14:46:11     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
14:46:11     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
14:46:11     at java.lang.reflect.Method.invoke (Method.java:498)
14:46:11     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
14:46:11     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
14:46:11     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
14:46:11     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
14:46:11 [ERROR] 
14:46:11 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
14:46:11 [ERROR] 
14:46:11 [ERROR] For more information about the errors and possible solutions, please read the following articles:
14:46:11 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
14:46:11 [ERROR] 
14:46:11 [ERROR] After correcting the problems, you can resume the build with the command
14:46:11 [ERROR]   mvn <goals> -rf :kie-wb-tests-rest
14:46:11 Build step 'Build dependent downstream repositories (for PR builds)' marked build as failure
14:46:11 Terminating xvnc.
14:46:11 $ vncserver -kill :25
14:46:12 Killing Xvnc process ID 10192
14:46:13 Xvnc seems to be deadlocked.  Kill the process manually and then re-run
14:46:13     /usr/bin/vncserver -kill :25
14:46:13 to clean up the socket files.
14:46:15 Recording test results
14:47:43 Skipping publisher since build result is FAILURE
14:47:43 Skipping publisher since build result is FAILURE
14:47:43 Archiving artifacts
14:49:38 Adding one-line test results to commit status...
14:49:38 Setting status of f25aa8e28c1b0b7f0ac0d0bb734c13036839c5bc to FAILURE with url https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/kie-wb-common-downstream-pullrequests/107/ and message: 'Build finished. 4778 tests run, 10 skipped, 0 failed.'
14:49:38 Using context: Linux - full downstream
14:49:39 [WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
14:49:39 Finished: FAILURE</pre>
</details>

hasys commented:
Jenkins retest please

romartin commented:
Jenkins execute full downstream build
